qutebrowser/extensions/loader.py

```diff
@@ -93,6 +102,23 @@ def walk_components() -> Iterator[ExtensionInfo]:
         yield ExtensionInfo(name=name)
 
 
+def _walk_pyinstaller() -> Iterator[ExtensionInfo]:
+    """Walk extensions when using PyInstaller.
+
+    See https://github.com/pyinstaller/pyinstaller/issues/1905
```

Comment:
Note to self, we still should try to figure out whether this has regressed.

Reply:
I don't think it has, but I'm not sure if we can test it on an old version because of the workarounds we had to do symlinking stuff around. I suppose this might be an issue with the upstream fix for that stuff breaking this use case.

This only started happening with https://github.com/qutebrowser/qutebrowser/pull/7803/commits/c0d31d399d31d3c45f47e375826ed87c16807edf where the data files changed to be under `/Frameworks/qutebrowser/`. That's the same path that the python files are reported to be under so the issue is that somehow the `/Frameworks/qutebrowser/` folder existing is causing importlib to use the file loader instead of the frozen ones. But it seems to work fine on Windows (TODO: triple check with https://github.com/qutebrowser/qutebrowser/actions/runs/5846907490), so it might be some mac specific workaround going wonky.

I tried to make an upstream test fail [here](https://github.com/toofar/pyinstaller/pull/1/files), but it didn't. I think the script in that test as the right structure (a folder with data files at the same level as a folder with python files). I might switch tack and try to do a minimal reproducer based on qutebrowser (including spec file).

Anyway, I don't intend to treat this oddity as blocking (I really want to see all the nightly tests working again!). So I'll probably merge this PR tomorrow.

Reply:
This might be fixed by https://github.com/pyinstaller/pyinstaller/pull/7885.

Reply:
Looks like that fixes it indeed. Data files and dynamic modules are both being loaded fine with no special casing for frozen builds from us! Yet another issue that gets solved with me doing nothing but procrastinating :joy: 

I just did a quick build to check. Will double check in more detail tomorrow.